### PR TITLE
chore(core, sequencer): add names to all snapshot tests

### LINKED ISSUE #1656: add snapshot names to all snapshot asserts
As noted [here](https://github.com/astriaorg/astria/pull/1647#discussion_r1799439267), many snapshots throughout the codebase do not have an associated name, and are hence based on line number. If assertions are moved around, then the snapshots will break, which should not happen. We should add names to all snapshot tests.

┆Issue Number: ENG-916

### PR DESCRIPTION
**Pull request by ethanoroshiba**
## Summary
Added names to all snapshot tests which did not have them.

## Background
If a certain test had multiple snapshot assertions, they would be named based on their ordering. Hence any reordering, removal, or addition of the snapshots would break the test, which should not be the case. This change aims to fix this problem as well as establish a best practice of naming snapshots moving forward.

## Changes
- Added names to all snapshot tests and regenerated all snapshots not within Conductor (due to code freeze)

## Testing
Passing all tests.

## Related Issues
closes #1656

### CODE REVIEW

**Review by noot — APPROVED** [review]

File: crates/astria-core/src/primitive/v1/snapshots/astria_core__primitive__v1__tests__compat_address.snap [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
 source: crates/astria-core/src/primitive/v1/mod.rs
-expression: "&bech32_address"
+expression: "&compat_address"
```
Comment (SuperFluffy) [inline crates/astria-core/src/primitive/v1/snapshots/astria_core__primitive__v1__tests__compat_address.snap:3]: And for this, add a bit `compat_bech32_non_m_address`

File: crates/astria-core/src/primitive/v1/mod.rs [hunk 2]
```diff
@@ -845,15 +845,15 @@ mod tests {
             .prefix(ASTRIA_ADDRESS_PREFIX)
             .try_build()
             .unwrap();
-        insta::assert_json_snapshot!(&main_address.to_raw());
+        insta::assert_json_snapshot!("main_address", &main_address.to_raw());
```
Comment (SuperFluffy) [inline crates/astria-core/src/primitive/v1/mod.rs:848]: Let's never name this `main_bech32m_address`

**Review by SuperFluffy — APPROVED** [review]
The changes make for much cleaner files

File: crates/astria-core/src/primitive/v1/mod.rs (hunk 2 above)
Comment (ethanoroshiba) [reply crates/astria-core/src/primitive/v1/mod.rs]: I'm assuming you meant to name it `main_bech32m_address`?

**Review by joroshiba — APPROVED** [review]

**Pull request merged**